### PR TITLE
Implement runtime-provided periodic timer input for source operators

### PR DESCRIPTION
**Pull request by phil-opp**
The periodic timer inputs allows operators to periodically produce values. To use the new input types, operators specify an input that maps to `dora/timer/secs/X` where `X` is the number of seconds between ticks. In addition to `secs`, the runtime also support `millis` for millisecond intervals.

The new dora timer nodes are visualized in the following way:

```mermaid
flowchart TB
  node_1[\node_1/]
  logger[/logger\]
subgraph python-operator [node]
  python-operator/op[op]
end
subgraph ___dora___ [dora]
  subgraph ___timer_timer___ [timer]
    dora/timer/secs/5[\secs/5/]
    dora/timer/millis/100[\millis/100/]
  end
end 
  python-operator/op -- counter as test --> logger
  dora/timer/secs/5 -- tick --> logger
  dora/timer/millis/100 -- dora_time --> python-operator/op
  node_1 -- data --> python-operator/op

```

TODO:

- [x] Implement parsing of new `dora/timer/...` inputs
- [x] Visualize the new timer inputs
- [x] Spawn a thread to send the timer tick messages

### CODE REVIEW

File: apis/rust/node/src/config.rs [hunk 1]
```diff
@@ -84,23 +86,62 @@ impl std::ops::Deref for DataId {
...
+
+    pub fn operator(&self) -> &Option<OperatorId> {
+        match self {
+            InputMapping::User(mapping) => &mapping.operator,
+            InputMapping::Timer { .. } => &None,
+        }
+    }
+}
+
+impl ToString for InputMapping {
```
Comment (haixuanTao) [inline apis/rust/node/src/config.rs:113]: I have read that implementing `std::fmt::Display` is the best practice and provide a larger scope than `ToString` but as we use it for serializing, maybe `ToString` is best. I would have probably implemented `Display` but you can leave it as is.

**Review by haixuanTao — APPROVED** [review]

File: apis/rust/node/src/config.rs (hunk 1 above)
Comment (phil-opp) [reply apis/rust/node/src/config.rs:113]: Oh yes, good catch! Implementing `Display` is definitely better, and it should also work nicely with serde.

Comment (phil-opp) [reply apis/rust/node/src/config.rs]: Fixed in void intermediate allocations by implementing Display instead of ToString

Comment (phil-opp) [reply apis/rust/node/src/config.rs]: Fixed in 38199b4a44414b3c635de572acf66627450f4fbb

**Pull request merged**